src/components/SecretsTable/SecretsTable.js

```diff
@@ -28,124 +39,162 @@ const {
-      annotations,
-      add: <Close />
+      annotations
     };
+
+    if (selectedNamespace === ALL_NAMESPACES) {
+      formattedSecret.namespace = secret.namespace;
+    }
+
+    return formattedSecret;
   });
 
-  return loading ? (
-    <DataTableSkeleton rowCount={1} columnCount={initialHeaders.length} />
-  ) : (
-    <DataTable
-      rows={secretsFormatted}
-      headers={initialHeaders}
-      isSortable
-      useZebraStyles
-      render={({ rows, headers, getHeaderProps }) => (
-        <TableContainer>
-          <Table className="secrets">
-            <TableHead className="header">
-              <TableRow>
-                {headers.map((header, i) => {
-                  const headerCell =
-                    headers.length !== i + 1 ? (
-                      <TableHeader
-                        {...getHeaderProps({ header })}
-                        className="cellText"
-                        key={header.key}
-                      >
-                        {header.header}
-                      </TableHeader>
-                    ) : (
-                      <TableHeader
-                        key={header.key}
-                        data-testid="addIcon"
-                        className="cellIcon"
-                        {...getHeaderProps({
-                          header,
-                          onClick: handleNew
-                        })}
-                      >
-                        {header.header}
-                      </TableHeader>
+  return (
+    <div className="secrets">
+      <DataTable
+        rows={secretsFormatted}
+        headers={initialHeaders}
+        isSortable
+        render={({
+          rows,
+          headers,
+          getHeaderProps,
+          getRowProps,
+          getSelectionProps,
+          getBatchActionProps,
+          selectedRows
+        }) => (
+          <TableContainer>
+            <div className="header">
+              <div className="header-title">
+                <h4 className="bx--data-table-header__title">Secrets</h4>
+              </div>
+              <TableToolbarContent>
+                <div className="add-div">
+                  <Button
+                    disabled={loading}
+                    id="add-btn"
+                    data-testid="addButton"
+                    onClick={handleNew}
+                  >
+                    <Add className="add-icon" />
+                    Add Secret
+                  </Button>
+                </div>
+              </TableToolbarContent>
+            </div>
+            <TableToolbar>
+              <TableBatchActions {...getBatchActionProps()}>
+                <TableBatchAction
+                  id="delete-btn"
+                  data-testid="deleteButton"
+                  renderIcon={Delete}
+                  onClick={() => {
+                    handleDelete(
+                      selectedRows.map(secret => ({
+                        namespace: secret.id.split(':')[0],
```

Comment:
again this secret should be an object containing the name and namespace values instead of doing string manipulation